app/mailers/twilio_sender.rb

```diff
@@ -2,23 +2,91 @@
 
 # TwilioSender: Methods to interact with Twilio REST API
 class TwilioSender
+  TWILIO_ERROR_CODES = {
+    invalid_to_number: '21211',
+    blocked_number: '21610',
+    invalid_number: '21614',
+    unsupported_region: '21408',
+    unreachable_unavailable: '30003',
+    unavailable_ineligible: '30004',
+    non_existent_or_off: '30005',
+    sms_ineligible: '30006',
+    carrier_filter: '30007',
+    unknown_error: '30008'
+  }.freeze
+
   @client = Twilio::REST::Client.new(ENV['TWILLIO_API_ACCOUNT'], ENV['TWILLIO_API_KEY'])
-  def self.send_sms(patient, contents)
+  def self.handle_twilio_error_codes(patient, error_code)
+    case error_code
+    # Invalid To Number https://www.twilio.com/docs/api/errors/21211
+    when TWILIO_ERROR_CODES[:invalid_to_number]
+      dispatch_errored_contact_history_items(patient, 'Invalid recipient phone number.')
+    # Blocked Number Error https://www.twilio.com/docs/api/errors/21610
+    when TWILIO_ERROR_CODES[:blocked_number]
+      dispatch_errored_contact_history_items(patient, 'Recipient phone number blocked communication with Sara Alert')
+      monitoree_number = Phonelib.parse(phone_numbers[:monitoree_number], 'US').full_e164
+      BlockedNumber.create(phone_number: monitoree_number) unless BlockedNumber.exists?(phone_number: monitoree_number)
```

Comment:
I benchmarked find_or_create_by against .exists? against 100,000 entries and found that there was no difference in speed. Will keep as-is because I think it's more clear especially since we are assigning the value returned by create (or alternatively find_or_create_by) to a variable.